### PR TITLE
Upstream sync January 2021

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

**Comment by peterbarker** [comment]
OK, so this external dialects stuff is confusing.  What makes them any different to e.g. `ardupilotmega.xml`? (except, perhaps, scale)....

File: message_definitions/v1.0/ASLUAV.xml [hunk 1]
```diff
@@ -270,5 +270,24 @@
       <field type="uint8_t" name="sinr_ecio">SINR (LTE) or ECIO (WCDMA) as reported by modem (unconverted)</field>
       <field type="uint8_t" name="rsrq">RSRQ (LTE only) as reported by modem (unconverted)</field>
     </message>
+    <message id="214" name="SATCOM_LINK_STATUS">
```
Comment (peterbarker) [inline message_definitions/v1.0/ASLUAV.xml:273]: More message ID conflicts.  This is not going to ease any sort of future integration.

File: message_definitions/v1.0/ASLUAV.xml [hunk 2]
```diff
@@ -270,5 +270,24 @@
...
+      <field type="uint64_t" name="timestamp" units="us">Timestamp</field>
+      <field type="uint64_t" name="last_heartbeat" units="us">Timestamp of the last successful sbd session</field>
+      <field type="uint16_t" name="failed_sessions">Number of failed sessions</field>
+      <field type="uint16_t" name="successful_sessions">Number of successful sessions</field>
+      <field type="uint8_t" name="signal_quality">Signal quality</field>
+      <field type="uint8_t" name="ring_pending">Ring call pending</field>
+      <field type="uint8_t" name="tx_session_pending">Transmission session pending</field>
+      <field type="uint8_t" name="rx_session_pending">Receiving session pending</field>
+    </message>
+    <message id="215" name="SENSOR_AIRFLOW_ANGLES">
```
Comment (peterbarker) [inline message_definitions/v1.0/ASLUAV.xml:284]: This looks like a different form of ArduPilot's `AOA_SSA` message (@samuelctabor).  Except a couple of bytes of waste (float means you can use NaN for invalid....)

I mean this is in a dialect and all, so it doesn't affect ArduPilot, but it does show mavlink has a ways to go as a standard.

File: .github/workflows/test_and_deploy.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          pip install future lxml
+      - name: Run deploy script
+        run: |
+          ./scripts/update_generated_repos.sh
```
Comment (tridge) [inline .github/workflows/test_and_deploy.yml:98]: what is this? probably should be removed

**Review by tridge — CHANGES_REQUESTED** [review]
also need to remove pymavlink change

File: .github/workflows/test_and_deploy.yml (hunk 3 above)
Comment (amilcarlucas) [reply .github/workflows/test_and_deploy.yml]: done

**Comment by amilcarlucas** [comment]
Changes done, and tests pass

File: message_definitions/v1.0/common.xml [hunk 4]
```diff
@@ -211,6 +211,9 @@
       <entry value="536870912" name="MAV_SYS_STATUS_OBSTACLE_AVOIDANCE">
         <description>0x20000000 Avoidance/collision prevention</description>
       </entry>
+      <entry value="1073741824" name="MAV_SYS_STATUS_SENSOR_PORPULSION">
+        <description>0x40000000 porpulsion (actuator, esc, motor or propellor)</description>
```
Comment (peterbarker) [inline message_definitions/v1.0/common.xml:215]: Spelling needs fixing.

That would be `PROPULSION`.

Comment (amilcarlucas) [reply message_definitions/v1.0/common.xml]: done

**Review by tridge — APPROVED** [review]

**Pull request merged**